### PR TITLE
Delivery for generate-a-piece-object

### PR DESCRIPTION
**Pull request by Lodigolden**
# Overview
Began work on Piece class

# Details
- Created class prototype
- Updated documentation

# Testing
UTs pass as expected: 
image

### CODE REVIEW

File: gaming-website/src/piece.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    }
+
+    ///////////////////////////////////////////////////////////////////////////
+    // Private member variables
+    ///////////////////////////////////////////////////////////////////////////
+    #row;       // Row the piece is in
+    #col;       // Col the piece is in
+    #is_king;   // Whether the piece is a king or not
+    #color;     // Color of the piece
+}
```
Comment (Lodigolden) [inline gaming-website/src/piece.js:40]:
```suggestion
};
```

File: gaming-website/tests/pieces.test.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        let game_piece = new Checkers_Piece();
+
+        // When - set_king() is called,
+        game_piece.set_king(false); 
+
+        // Then - get_king() is correct.
+        expect(game_piece.get_king()).toBe(false); 
+    })
+
+    ///////////////////////////////////////////////////////////////////////////
```
Comment (Lodigolden) [inline gaming-website/tests/pieces.test.js:22]: Delete this line until more UTs are added.

File: gaming-website/src/piece.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
+export class Checkers_Piece
```
Comment (Lodigolden) [inline gaming-website/src/piece.js:1]: Add description at top of class.

File: gaming-website/src/piece.js (hunk 1 above)
Comment (Lodigolden) [inline gaming-website/src/piece.js:40]:
```suggestion
};
```

File: gaming-website/tests/pieces.test.js (hunk 2 above)
Comment (Lodigolden) [inline gaming-website/tests/pieces.test.js:22]: Delete this line until more UTs are added.

File: gaming-website/src/piece.js (hunk 3 above)
Comment (Lodigolden) [reply gaming-website/src/piece.js]: - [ ] Add description

File: gaming-website/src/piece.js (hunk 1 above)
Comment (Lodigolden) [reply gaming-website/src/piece.js]: - [ ] Fix change

Comment (Lodigolden) [reply gaming-website/src/piece.js]: **DONE**

Comment (Lodigolden) [reply gaming-website/src/piece.js]: **VERIFIED**

File: gaming-website/tests/pieces.test.js (hunk 2 above)
Comment (Lodigolden) [reply gaming-website/tests/pieces.test.js]: **DONE**

Comment (Lodigolden) [reply gaming-website/tests/pieces.test.js]: **VERIFIED**

File: gaming-website/src/piece.js (hunk 3 above)
Comment (Lodigolden) [reply gaming-website/src/piece.js]: **VERIFIED**

**Review by Alexcburn — APPROVED** [review]

**Pull request merged**